docs/how-to-guides/contributing.rst

```diff
@@ -0,0 +1,86 @@
+.. SPDX-FileCopyrightText: 2021 The meson-python developers
+..
+.. SPDX-License-Identifier: MIT
+
+.. _how-to-guides-contributing:
+
+Contributing
+============
+
+Thank you for your interest in contributing.  ``meson-python`` development is
+hosted on GitHub_ and uses the ubiquitous pull-request based workflow.  The
+usual Python development practices apply.  Python source code should adhere to
+the `PEP 8`_ code style with a maximum line length of 127 characters.  Single
+quotes are used as string delimiters.  Python code uses typing annotations
+verified via Mypy_. Python code is kept lint-free sweeping it with Ruff_.
+
+``meson-python`` focus is producing Python wheels containing extension modules
+and thus running its test suite requires C and C++ compilers, and CMake. The
```

Comment:
CMake? I guess you meant Ninja?

Reply:
`ninja` from PyPI seems fine indeed, it's small enough. The advanced contributor will understand anyway that system Ninja is fine too.